plugins/icesat2/plugin/Atl03Reader.cpp

```diff
@@ -280,7 +280,15 @@ Atl03Reader::Region::Region (info_t* info):
         }
         else
         {
-            return; // early exit since no subsetting required
+            num_segments = segment_ph_cnt.size;
+            if(num_segments > 0)
```

Comment:
I think this check is superfluous since the for loop will immediately exit if num_segments is negative